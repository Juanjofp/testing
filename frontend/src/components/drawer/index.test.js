import React from 'react';
import ReactDOM from 'react-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import {DrawerMenu} from './index';

describe(
    'DrawerMenu',
    () => {
        it(
            'renders without crashing', 
            () => {
                const div = document.createElement('div');
                ReactDOM.render(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={true}
                                closeDrawer={()=>{}}
                                isLogin={false}
                                isAdmin={false}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>,
                    div
                );
            }
        );

        it(
            'renders shallow without crash',
            () => {
                const wraper = shallow(
                    <DrawerMenu
                        isOpen={true}
                        closeDrawer={()=>{}}
                        isLogin={false}
                        isAdmin={false}
                        location={{pathname: '/'}}/>
                );
            }
        );

        it(
            'create an snapshot for drawer',
            () => {
                const tree = render(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={false}
                                closeDrawer={()=>{}}
                                isLogin={false}
                                isAdmin={false}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>
                );
                expect(tree).toMatchSnapshot();
            }
        );

        it(
            'when is Closed do not render Menuitems',
            () => {
                const wraper = mount(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={false}
                                closeDrawer={()=>{}}
                                isLogin={false}
                                isAdmin={false}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>
                );
                const isOpen = wraper.find('DrawerMenu').props().isOpen;
                expect(isOpen).toBe(false);
            }
        );

        it(
            'when is Anonymous Renders 3 Menuitems',
            () => {
                const wraper = mount(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={true}
                                closeDrawer={()=>{}}
                                isLogin={false}
                                isAdmin={false}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>
                );
                const menuItems = wraper.find('MenuItem');
                expect(menuItems.length).toBe(3);
                expect(menuItems.first().contains('HOME')).toBe(true);
                expect(menuItems.at(2).contains('LOGIN')).toBe(true);
            }
        );

        it(
            'when is Login Renders 3 Menuitems',
            () => {
                const wraper = mount(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={true}
                                closeDrawer={()=>{}}
                                isLogin={true}
                                isAdmin={false}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>
                );
                const menuItems = wraper.find('MenuItem');
                expect(menuItems.length).toBe(3);
                expect(menuItems.first().contains('HOME')).toBe(true);
                expect(menuItems.at(2).contains('CLOSE')).toBe(true);
            }
        );

        it(
            'when is Admin Renders 4 Menuitems',
            () => {
                const wraper = mount(
                    <MuiThemeProvider>
                        <MemoryRouter>
                            <DrawerMenu
                                isOpen={true}
                                closeDrawer={()=>{}}
                                isLogin={true}
                                isAdmin={true}
                                location={{pathname: '/'}}/>
                        </MemoryRouter>
                    </MuiThemeProvider>
                );
                const menuItems = wraper.find('MenuItem');
                expect(menuItems.length).toBe(4);
                expect(menuItems.first().contains('HOME')).toBe(true);
                expect(menuItems.at(3).contains('CLOSE')).toBe(true);
            }
        );
    }
);
