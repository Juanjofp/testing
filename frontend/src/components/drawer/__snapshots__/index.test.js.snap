// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu create an snapshot for drawer 1`] = `
<div>
  <div
    style="position:fixed;height:100%;width:100%;top:0;left:-100%;opacity:0;background-color:rgba(0, 0, 0, 0.54);-webkit-tap-highlight-color:rgba(0, 0, 0, 0);will-change:opacity;transform:translateZ(0);transition:left 0ms cubic-bezier(0.23, 1, 0.32, 1) 400ms, opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;z-index:1200;pointer-events:none;mui-prepared:"
  >
    
  </div>
  <div
    style="color:rgba(0, 0, 0, 0.87);background-color:white;transition:transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 3px 10px rgba(0, 0, 0, 0.16),
           0 3px 10px rgba(0, 0, 0, 0.23);border-radius:0px;height:100%;width:256px;position:fixed;z-index:1300;left:0;top:0;transform:translate(-266px, 0);overflow:auto;-webkit-overflow-scrolling:touch;mui-prepared:"
  >
    <div
      style="color:rgba(0, 0, 0, 0.87);background-color:#00838F;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23);border-radius:2px;padding:20px;mui-prepared:"
    >
      <img
        alt="jhon@doe"
        data-testid="drawer-avatar"
        size="80"
        src="https://quizup-players.imgix.net/avatars/Avatar_astronaut-large-hd.png"
        style="color:#ffffff;background-color:rgb(188, 188, 188);user-select:none;display:inline-flex;align-items:center;justify-content:center;font-size:40px;border-radius:50%;height:80px;width:80px;mui-prepared:"
      >
        
      </img>
      <div
        style="color:white"
      >
        jhon@doe
      </div>
    </div>
    <hr
      style="margin:0;margin-top:-1px;margin-left:0;height:1px;border:none;background-color:#e0e0e0;mui-prepared:"
    >
      
    </hr>
    <div>
      <a
        exact="true"
        href="/"
        role="menuitem"
        style="border:10px;box-sizing:border-box;display:block;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);cursor:pointer;text-decoration:none;margin:0;padding:0;outline:none;font-size:16px;font-weight:inherit;position:relative;color:rgba(0, 0, 0, 0.87);line-height:48px;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;min-height:48px;white-space:nowrap;background:none;mui-prepared:"
        tabindex="0"
      >
        <div>
          <div
            style="margin-left:0;padding-left:72px;padding-right:72px;padding-bottom:0;padding-top:0;position:relative;mui-prepared:"
          >
            <svg
              style="display:block;color:rgba(0, 0, 0, 0.87);fill:#757575;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;position:absolute;top:0;margin:12px;left:4px;mui-prepared:"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              >
                
              </path>
            </svg>
            <svg
              style="display:block;color:rgba(0, 0, 0, 0.87);fill:#757575;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;position:absolute;top:0;margin:12px;right:4px;mui-prepared:"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
              >
                
              </path>
            </svg>
            <div
              style="mui-prepared:"
            >
              HOME
            </div>
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/private"
        role="menuitem"
        style="border:10px;box-sizing:border-box;display:block;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);cursor:pointer;text-decoration:none;margin:0;padding:0;outline:none;font-size:16px;font-weight:inherit;position:relative;color:rgba(0, 0, 0, 0.87);line-height:48px;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;min-height:48px;white-space:nowrap;background:none;mui-prepared:"
        tabindex="0"
      >
        <div>
          <div
            style="margin-left:0;padding-left:72px;padding-right:16px;padding-bottom:0;padding-top:0;position:relative;mui-prepared:"
          >
            <svg
              style="display:block;color:rgba(0, 0, 0, 0.87);fill:#757575;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;position:absolute;top:0;margin:12px;left:4px;mui-prepared:"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              >
                
              </path>
            </svg>
            <div
              style="mui-prepared:"
            >
              PRIVATE
            </div>
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/login"
        role="menuitem"
        style="border:10px;box-sizing:border-box;display:block;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);cursor:pointer;text-decoration:none;margin:0;padding:0;outline:none;font-size:16px;font-weight:inherit;position:relative;color:rgba(0, 0, 0, 0.87);line-height:48px;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;min-height:48px;white-space:nowrap;background:none;mui-prepared:"
        tabindex="0"
      >
        <div>
          <div
            style="margin-left:0;padding-left:72px;padding-right:16px;padding-bottom:0;padding-top:0;position:relative;mui-prepared:"
          >
            <svg
              style="display:block;color:rgba(0, 0, 0, 0.87);fill:#757575;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;position:absolute;top:0;margin:12px;left:4px;mui-prepared:"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              >
                
              </path>
            </svg>
            <div
              style="mui-prepared:"
            >
              LOGIN
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
